heroku/deploy-a-django-app.md

```diff
@@ -28,24 +28,282 @@ preferred platform for hosting dynamic applications.
+To create these resources, start by defining the name of your app:
+
+```bash
+# This should be the same slug as your project's GitHub repo.
+export APP_NAME=<your-app-name-here>
+```
+
+Then, run the following Heroku CLI commands to create a staging app and a pipeline:
+
+```bash
+heroku create ${APP_NAME}-staging -t datamade --manifest
+heroku pipelines:create -t datamade ${APP_NAME} -a ${APP_NAME}-staging -s staging
+heroku pipelines:connect ${APP_NAME} -r datamade/${APP_NAME}
+```
+
+Your CLI output should look like this:
+```bash
+heroku create ${APP_NAME}-staging -t datamade --manifest 
+Reading heroku.yml manifest... done
+Creating ⬢ demo-app-staging... done, stack is container
+Adding heroku-postgresql... done
+https://demo-app-staging.herokuapp.com/ | https://git.heroku.com/demo-app-staging.git
+
+heroku pipelines:add ${APP_NAME}-staging -a ${APP_NAME}-staging -s staging
+Adding ⬢ demo-app-staging to datamade-app pipeline as staging... done
+```
+
+If you would like to set up a production app as well, run the following commands
+to create one and add it to your pipeline:
+
+```bash
+heroku create ${APP_NAME} -t datamade --manifest
+heroku pipelines:add ${APP_NAME} -a ${APP_NAME} -s production
+```
+
+Once you have the environments you need, enable [review
+apps](https://devcenter.heroku.com/articles/github-integration-review-apps)
+for your pipeline:
+
+```bash
+# Note that these need to be two separate commands due to an open Heroku bug,
+# since --autodeploy and --autodestroy require a PATCH request
+heroku reviewapps:enable -p ${APP_NAME}
+heroku reviewapps:enable -p ${APP_NAME} --autodeploy --autodestroy
+```
+
+Next, configure environment variables for staging and production apps. `DJANGO_SECRET_KEY`
+should be a string generated using the [XKCD password generator](https://preshing.com/20110811/xkcd-password-generator/),
+and `DJANGO_ALLOWED_HOSTS` should be a comma-separated string of valid hosts
+for your app.
+
+```bash
+heroku config:set -a ${APP_NAME}-staging DJANGO_SECRET_KEY=<random-string-here>
+heroku config:set -a ${APP_NAME}-staging DJANGO_ALLOWED_HOSTS=.herokuapp.com
+
+# Run these commands if you have a production application
+heroku config:set -a ${APP_NAME} DJANGO_SECRET_KEY=<random-string-here>
+heroku config:set -a ${APP_NAME} DJANGO_ALLOWED_HOSTS=<your-list-of-allowed-hosts-here>
+```
+
+Note that review app config vars cannot yet be set using the CLI, but you can set them in
+the Heroku dashboard by navigating to the pipeline home page and visiting
+`Settings > Review Apps > Review app config vars` in the nav.
+
+Also note that while `DATABASE_URL` is probably required by your application, you don't actually
+need to set it yourself. The Heroku Postgres add-on will [automatically define this
+variable](https://devcenter.heroku.com/articles/heroku-postgresql#designating-a-primary-database)
+when it provisions a database for your application.
+
```

Comment:
This could use a subhed, as well. `Configure deployments from Git branches`?

Reply:
This change is in the above commit, too: 9c03a68